tests/optimize/test_adaptive.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the ``AdaptiveOptimizer``.
```

Comment:
These tests are good, but I'm concerned we're missing some more fundamental tests. I'm thinking:
- Test that the gate with the largest gradient is selected
- Test that pool is properly "drained" if there are repeated gates (and not drained if not)
- Test that the `append_gate` method works as intended
- Test that outputs are correct by comparing to cases where there is an analytical answer, e.g., a simple single-qubit circuit

Reply:
> Test that the gate with the largest gradient is selected
Added.

> Test that pool is properly "drained" if there are repeated gates (and not drained if not)
Do you mean checking the input pool and remove redundant gates somehow?

> Test that the append_gate method works as intended
Added.

> Test that outputs are correct by comparing to cases where there is an analytical answer, e.g., a simple single-qubit circuit
Added a qubit rotation example with trivial answers.